File: core/Entities/Basket.cs
namespace core.Entities
{
    public class Basket
    {
        public Basket()
        {
        }

        public Basket(string id)
        {
            Id = id;
        }

        public string Id { get; set; }
        public List<Item> Items { get; set; } = new List<Item>();
    }
}

File: client/src/app/shared/models/basket.ts
import { v4 as uuidv4 } from 'uuid';

export interface IBasket {
  id: string;
  items: IItem[];
}

export interface IItem {
  id: number;
  name: string;
  price: number;
  quantity: number;
  pictureUrl: string;
  brand: string;
  type: string;
}

export class Basket implements IBasket {
  id = uuidv4();
  items: IItem[] = [];
}

export interface IBasketTotal {
  amount: number;
  subtotal: number;
  total: number;
}
